src/core/server/index.test.ts

```diff
@@ -41,88 +42,86 @@ import { BehaviorSubject } from 'rxjs';
 import { Server } from '.';
 import { Env } from './config';
 import { getEnvOptions } from './config/__mocks__/env';
-import { logger } from './logging/__mocks__';
+import { loggingServiceMock } from './logging/logging_service.mock';
+
+import { configServiceMock } from './config/config_service.mock';
 
-const mockConfigService = { atPath: jest.fn(), getUnusedPaths: jest.fn().mockReturnValue([]) };
+const configService = configServiceMock.create();
 const env = new Env('.', getEnvOptions());
+const logger = loggingServiceMock.create();
 
 beforeEach(() => {
-  mockConfigService.atPath.mockReturnValue(new BehaviorSubject({ autoListen: true }));
+  configService.atPath.mockReturnValue(new BehaviorSubject({ autoListen: true }));
 });
 
 afterEach(() => {
-  logger.mockClear();
-  mockConfigService.atPath.mockReset();
-  mockHttpService.start.mockReset();
-  mockHttpService.stop.mockReset();
-  mockElasticsearchService.start.mockReset();
-  mockElasticsearchService.stop.mockReset();
+  jest.clearAllMocks();
+
+  configService.atPath.mockReset();
+  httpService.start.mockReset();
+  httpService.stop.mockReset();
+  elasticsearchService.start.mockReset();
+  elasticsearchService.stop.mockReset();
   mockPluginsService.start.mockReset();
   mockPluginsService.stop.mockReset();
   mockLegacyService.start.mockReset();
   mockLegacyService.stop.mockReset();
 });
 
 test('starts services on "start"', async () => {
-  const mockHttpServiceStart = { something: true };
-  mockHttpService.start.mockReturnValue(Promise.resolve(mockHttpServiceStart));
-
-  const mockElasticsearchServiceStart = { adminClient$: {} };
-  mockElasticsearchService.start.mockResolvedValue(mockElasticsearchServiceStart);
-
   const mockPluginsServiceStart = new Map([['some-plugin', 'some-value']]);
   mockPluginsService.start.mockReturnValue(Promise.resolve(mockPluginsServiceStart));
 
-  const server = new Server(mockConfigService as any, logger, env);
+  const server = new Server(configService as any, logger, env);
```

Comment:
here we have to suppress ts errors because`Server` declared a dependency on `ConfigService` class, not its interface. An alternative approach could be
```js
// config.js
class ConfigService implements IConfigService {...
// server.js
class Server {
  constructor(configService: IConfigService, ...)
```
it only complicates the code tho

Reply:
Interesting... in other classical OOO languages I've worked with, this issue is resolved by having the mock extend the original base class. Is it possible to define a base class or something with jest mocks?

Reply:
Sound interesting, but seems it is not something that jest does out-of-the-box https://jestjs.io/docs/en/es6-class-mocks

even if it was possible, imho this makes sense with abstract classes, but with real ones it's painful. 
let's assume we want to extend a real class
```typescript
class ServiceMock extends Service {
  // [1]
  constructor() {
    super({} as any)
  }
  // [2]
  public getFoo() {
    return null;
  }
   // [3]
  public start() {
    return Promise.resolve({....});
  }
}
```
[1] we **have to** call parent class constructor. not the thing we want to do in tests.
[2] mock implementation is easily out of sync with the parent class.
It could contain methods aren't defined in the parent class.
Without using `abstract class` we cannot force derived class to implement all parent class methods.
[3] we still need to find how to add mocks for public methods.

Reply:
That's a shame, but it makes sense. I'm down with exposing interfaces for services, though I'm not a fan of hungarian notation. I don't have an alternative naming recommendation at the moment, but I'll think about it.

I'm fine moving forward with this PR as-is, if you are.

Reply:
yes, that was a note for further discussion 👍